cli/ballerina-cli-utils/src/main/ballerina/packaging.pull/ballerina.pull.bal

```diff
@@ -53,35 +59,26 @@ function pullPackage (string url, string destDirPath, string fullPkgPath, string
             error err = {message:"package size information is missing from the remote repository"};
             throw err;
         }
-        int pkgSize;
-        var conversion = <int> contentLengthHeader;
-        match conversion{
-            error conversionErr => throw conversionErr;
-            int size => pkgSize = size;
-        }
-        io:ByteChannel sourceChannel = {};
-        var srcChannel = res.getByteChannel();
-        match srcChannel {
-            mime:EntityError errRes => {
-                var errorResp = <error> errRes;
-                match errorResp {
-                    error err =>  throw err;
-                }
-            }
-            io:ByteChannel channel => sourceChannel = channel;
-        }
-
-        // Get the package version from the canonical header of the response
-        string linkHeaderVal;
-        if (res.hasHeader("Link")) {
-            linkHeaderVal = res.getHeader("Link");
-        } else {
-            error err = {message:"package version information is missing from the remote repository"};
-            throw err;
-        }
-
-        string canonicalLinkURL = linkHeaderVal.subString(linkHeaderVal.indexOf("<") + 1, linkHeaderVal.indexOf(">"));
-        string pkgVersion = canonicalLinkURL.subString(canonicalLinkURL.lastIndexOf("/") + 1, canonicalLinkURL.length());
+        int pkgSize = check <int> contentLengthHeader;
+    
+        io:ByteChannel sourceChannel = check (res.getByteChannel());
+    
+        string rawPathVal;
+        if (res.hasHeader("raw-path")) {
```

Comment:
raw-path in not very nice, we'll have to see if we can get this info from the request.